c++/src/kj/async-inl.h

```diff
@@ -1095,7 +1114,33 @@ template <typename Func>
 inline PromiseForResult<Func, WaitScope&> startFiber(size_t stackSize, Func&& func) {
   typedef _::FixVoid<_::ReturnType<Func, WaitScope&>> ResultT;
 
-  Own<_::FiberBase> intermediate = kj::heap<_::Fiber<Func>>(stackSize, kj::fwd<Func>(func));
+  Own<_::FiberStack> stack = kj::heap<_::FiberStack>(stackSize);
```

Comment:
Ideally, we want to keep the amount of code in a template function as minimal as possible, since the compiler generates a separate copy of this whole function for each instantiation (which, in this function's case, is each place where it is called, since the `Func` is different every time!).

Instead of adding all this stack-allocating code here, what if `FiberBase`'s constructor accepted `size_t stackSize, Maybe<FiberPool&>`, and then did the appropriate allocation depending on whether a FiberPool was provided?

I think this would also allow you to move `FiberStack`'s definition entirely to the `.c++` file, which is always nice since the header file needs to be re-parsed for every single dependent file we compile.

Reply:
Or if it was a separate constructor, yeah I could see that.